lib/kubeclient.rb

```diff
@@ -704,7 +705,6 @@ def return_or_yield_to_watcher(watcher, &block)
     end
 
     def http_options(uri)
-      refresh_bearer_token_from_file
```

Comment:
What about `watch_foo` connections?
These don't use faraday yet (pending #488) but `http` gem.  
IIUC #532 ensured watch requests get a fresh token by this line (the name `http_options` literally means "options for `http` gem) and I think dropping this means they won't get fresh tokens, or perhaps won't get a bearer token _at all_?

Reply:
@cben - I'll have a look at watch_*, and unbreak that if I did break it.

Reply:
watch_foo was broken, both here and on master. It should be less broken now, as we ram through the authorization header from either :bearer_token or :bearer_token_file